src/creatures/players/player.cpp

```diff
@@ -790,10 +793,16 @@ bool Player::canWalkthrough(const Creature* creature) const
 
 		thisPlayer->setLastWalkthroughPosition(creature->getPosition());
 		return true;
+	// Lua npcs
 	} else if (npc) {
-		const Tile* tile = npc->getTile();
-		const HouseTile* houseTile = dynamic_cast<const HouseTile*>(tile);
-		return (houseTile != nullptr);
+		const Tile* npcTile = npc->getTile();
+		const HouseTile* npcHouseTile = dynamic_cast<const HouseTile*>(npcTile);
+		return (npcHouseTile != nullptr);
+	// XML npcs
+	} else if (npcOld) {
+		const Tile* npcOldtile = npcOld->getTile();
+		const HouseTile* npcOldHouseTile = dynamic_cast<const HouseTile*>(npcOldtile);
+		return (npcOldHouseTile != nullptr);
```

Comment:
those two else ifs are exactly the same thing, you can have an interface that both Npc's extend and you receive this interface instead, than you don't need to repeat code here.